docs/template_plugin/tests/functional/subgraph_reference/preprocess.cpp

```diff
@@ -419,6 +419,113 @@ static RefPreprocessParams resize_lvalues() {
     return res;
 }
 
+static RefPreprocessParams convert_layout_nhwc_to_nchw_shape_lvalue() {
+    RefPreprocessParams res("convert_layout_nhwc_to_nchw_shape_lvalue");
+    res.function = []() {
+        auto f = create_simple_function(element::u8, {1, 3, 2, 2});
+        f->get_parameters()[0]->set_layout("NCHW");
+        auto t = InputTensorInfo();
+        t.set_layout("NHWC");
+        t.set_data_shape({1, 2, 2, 3});
+        f = PrePostProcessor()
+                .input(InputInfo()
+                               .tensor(std::move(t))
+                               .preprocess(PreProcessSteps().convert_layout("NCHW")))
+                .build(f);
+        return f;
+    };
+    res.inputs.emplace_back(Shape{1, 2, 2, 3}, element::u8, std::vector<uint8_t>{1,  2,  3,       // [H=0, W=0, RGB]
+                                                                                 4,  5,  6,       // [H=0, W=1]
+                                                                                 7,  8,  9,       // [H=1, W=0]
+                                                                                 10, 11, 12});    // [H=1, W=1]
+    res.expected.emplace_back(Shape{1, 3, 2, 2}, element::u8, std::vector<uint8_t>{1, 4, 7, 10,    // R
+                                                                                   2, 5, 8, 11,    // G
+                                                                                   3, 6, 9, 12});  // B
+    return res;
+}
+
+static RefPreprocessParams convert_layout_nhwc_to_nchw_shape() {
+    RefPreprocessParams res("convert_layout_nhwc_to_nchw_shape");
+    res.function = []() {
+        auto f = create_simple_function(element::u8, {1, 3, 2, 2});
+        f->get_parameters()[0]->set_layout("NCHW");
+        f = PrePostProcessor()
+                .input(InputInfo()
+                               .tensor(InputTensorInfo().set_layout("NHWC").set_data_shape({1, 2, 2, 3}))
```

Comment:
personally, I don't want to have `set_data_shape`

It has a lot of confusions with `set_static_spatial_dims`

Reply:
Well it's not ideal, but seems like in general we'll have to provide such API to user to set shape of their input data (like layout and element type). As I wrote in description for `set_data_shape`, `In many cases tensor's shape is calculated based on layout, color format, spatial dimensions, however user has possibility to manually set of input shape.`
In general it looks not possible to calculate user's input shape, especially with custom operations and considering partial layouts

Reply:
> however user has possibility to manually set of input shape.

has because you are adding this function which does not seem necessary right now.
Can you provide example when we need it?

Reply:
BTW, is this method needed for current PR? In no, can we skip it and merge at least fully specified layouts conversion?

Reply:
ok, agree, removed 'set_data_shape'